ports/atmel-samd/boards/hallowing_m0_express/board.c

```diff
@@ -26,13 +26,78 @@
 
 #include "boards/board.h"
 
-void board_init(void)
-{
+#include "shared-bindings/displayio/FourWire.h"
+
+#include "tick.h"
+
+displayio_fourwire_obj_t board_display_obj;
+
+uint8_t display_init_sequence[] = {
+    0x01, 0, // SWRESET
+    0x11, 0, // SLPOUT
+    0xb1, 3, 0x01, 0x2C, 0x2D, // _FRMCTR1
+    0xb2, 3, 0x01, 0x2C, 0x2D, //
+    0xb3, 6, 0x01, 0x2C, 0x2D, 0x01, 0x2C, 0x2D,
+    0xb4, 1, 0x07, // _INVCTR line inversion
+    0xc0, 3, 0xa2, 0x02, 0x84, // _PWCTR1 GVDD = 4.7V, 1.0uA
+    0xc1, 1, 0xc5, // _PWCTR2 VGH=14.7V, VGL=-7.35V
+    0xc2, 2, 0x0a, 0x00, // _PWCTR3 Opamp current small, Boost frequency
+    0xc3, 2, 0x8a, 0x2a,
+    0xc4, 2, 0x8a, 0xee,
+    0xc5, 1, 0x0e, // _VMCTR1 VCOMH = 4V, VOML = -1.1V
+    0x2a, 0, // _INVOFF
+    0x36, 1, 0x18, // _MADCTL bottom to top refresh
+    // 1 clk cycle nonoverlap, 2 cycle gate rise, 3 sycle osc equalie,
+    // fix on VTL
+    0x3a, 1, 0x05, // COLMOD - 16bit color
+    0xe0, 16, 0x02, 0x1c, 0x07, 0x12, // _GMCTRP1 Gamma
+              0x37, 0x32, 0x29, 0x2d,
+              0x29, 0x25, 0x2B, 0x39,
+              0x00, 0x01, 0x03, 0x10,
+    0xe1, 16, 0x03, 0x1d, 0x07, 0x06, // _GMCTRN1
+              0x2E, 0x2C, 0x29, 0x2D,
+              0x2E, 0x2E, 0x37, 0x3F,
+              0x00, 0x00, 0x02, 0x10,
+    0x2a, 3, 0x02, 0x00, 0x81, // _CASET XSTART = 2, XEND = 129
+    0x2b, 3, 0x02, 0x00, 0x81, // _RASET XSTART = 2, XEND = 129
+    0x13, 0, // _NORON
+    0x29, 0, // _DISPON
+};
+
+void board_init(void) {
+    board_display_obj.base.type = &displayio_fourwire_type;
+    common_hal_displayio_fourwire_construct(&board_display_obj,
+        &pin_PB23, &pin_PB22, &pin_PA28, &pin_PA01, &pin_PA27,
+        128, 128, 2, 0, 16, 0x2a, 0x2b, 0x2c);
```

Comment:
constants would probably be a cleaner way

Reply:
I think those are the least confusing of the whole constructor, I meant more for the pins it would be nice to know which is DIN, CLK, DC, CS and what the 2, 0, 16 mean.